## INCLASS LAB-EXERCISE DAY 02 Questions

-----------

In [1]:
## CALLING LIBRARIES

import numpy as np
import pandas as pd
import scipy.stats as stats
import math
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import scipy.stats as skewnorm

### Sampling and Confidence Interval

**1. Problem:**).What is the sampling error and standard error for the given the population data and sample data.Explain what is sampling error and how can standard error can be reduced.**

**Population data.**

In [4]:
Marks_scored=[60,80,70,75,80,90,100,100,65,75,60,90,70,70,80,50,55,60,70,65,70,55,50,65,70,65,70,65,75,80,90,95,95,85,90,80,75,70,80,85,90,80,75,70,60,65,70,80,75,80,90,65,80,65,70,85,75,70,80,85,65,70,85,80,65,70,80,90,100,60,70,55,60,70,70,70,80,85,90,80,75,70,60,65,70,80,75,80,90,70,85,80,65,70,80,90,100,60,70,80]

**Sample data.**

In [5]:
sam_marks=[90,100,100,65,75,60,90,70,65,70,65,70,65,60,70,85,65,70,85,80,65,70,80,65,70]

**Solution:**

In [10]:
po_mean=np.mean(Marks_scored)
sam_mean=np.mean(sam_marks)
po_mean-sam_mean
po_std=np.std(Marks_scored)

----------------------

**2. Problem:** Provided the Population data and sample data.**

* a).plot the population data and sample data check how the data is distributed.Give inference wheather the sampling distribution similar to population data.

In [2]:
np.random.seed(10)
population_ages1 = stats.poisson.rvs(loc=18, mu=35, size=150000)
population_ages2 = stats.poisson.rvs(loc=18, mu=10, size=100000)
population_ages = np.concatenate((population_ages1, population_ages2))


print(population_ages1)

[58 55 53 ... 50 46 49]


In [4]:

np.random.seed(6)
sample_ages = np.random.choice(a= population_ages,size=500)            # Sample 500 values
print(sample_ages)

[23 53 52 24 52 29 47 25 35 32 48 21 54 50 25 55 51 28 58 56 46 49 27 63
 54 45 49 27 50 58 30 51 50 44 28 30 28 46 49 28 51 61 55 49 26 27 52 53
 45 47 26 58 50 62 60 49 29 26 60 53 25 29 54 30 56 28 34 60 46 24 31 53
 60 59 64 57 52 49 29 27 44 68 27 55 66 58 51 39 57 48 47 30 26 51 53 55
 31 25 25 41 52 27 57 59 31 29 47 51 53 28 62 28 29 52 58 54 50 31 46 31
 54 28 52 41 46 47 30 29 49 45 23 50 26 50 48 26 34 49 26 31 27 31 26 61
 26 29 28 25 46 33 48 56 51 42 30 30 30 31 39 66 56 42 24 47 33 30 58 63
 29 60 25 46 41 52 47 49 23 27 53 48 27 50 58 27 30 51 56 54 43 28 60 44
 25 27 21 53 32 63 50 52 53 29 55 49 29 56 47 28 59 33 31 61 32 65 27 27
 26 26 39 60 26 51 25 51 28 58 46 28 28 56 28 60 53 56 44 63 25 51 44 47
 56 43 57 23 29 53 28 54 28 26 57 54 46 54 26 53 28 58 31 50 54 25 49 29
 27 60 28 50 30 52 26 62 33 28 57 52 27 57 46 46 31 56 27 51 24 51 46 23
 24 30 26 54 27 27 25 55 34 30 50 55 59 58 34 34 29 26 45 50 49 56 29 54
 52 52 23 36 45 29 59 47 27 47 48 46 65 55 31 30 53

**Solution :**

In [10]:
pop_mean=np.mean(population_ages1)--53.00546
samp_mean=np.mean(sample_ages)--42.388
print('Sample error :' ,np.abs(samp_mean-pop_mean))

Sample error : 21.234920000000002


-------------------

**3. Problem:** Provided the Population data and sample data.

* a).calculate a 95% confidence for our mean point estimate:
    

In [29]:
np.random.seed(10)
population_ages1 = stats.poisson.rvs(loc=18, mu=35, size=150000)
population_ages2 = stats.poisson.rvs(loc=18, mu=10, size=100000)
population_ages = np.concatenate((population_ages1, population_ages2))

population_ages #Population data

array([58, 55, 53, ..., 27, 28, 28])

In [31]:
np.random.seed(10)

sample_size = 1000
sample = np.random.choice(a= population_ages, size = sample_size)
sample_mean = sample.mean()

**solution :**

In [49]:
print('No of observations : ',len(sample))
print('Sample Mean : ',np.mean(sample))
print('Standard Mean : ',np.std(sample,ddof=1))
ULT,LLT=stats.norm.interval(0.95,np.mean(sample),np.std(sample,ddof=1)/np.sqrt(len(sample)))
print('Upper Limit : ',ULT,' Lower Limit:',LLT)
z_alpha=np.abs(round(stats.norm.isf(q=0.1/2),4))
print('Z alpha :',z_alpha)
ME=(z_alpha*np.std(sample,ddof=1))/np.sqrt(len(sample))
print('Mean Error :',ME)
std=np.std(sample,ddof=1)
#From Mean Error calculating sample size
(z_alpha**2)*std**2/(ME**2)

No of observations :  1000
Sample Mean :  42.523
Standard Mean :  13.439273979091471
Upper Limit :  41.690040474997915  Lower Limit: 43.35595952500209
Z alpha : 1.6449
Mean Error : 0.699061377394371


1000.0000000000001

In [28]:
weight = [59.1, 65.0, 75.8, 79.2, 95.0, 99.8, 89.1, 65.3, 41.9, 55.2, 94.8, 84.1, 83.2, 74.0, 75.5, 76.2, 79.1, 80.1, 
          92.1, 74.2, 59.2, 64.0, 75, 78.2, 95.6, 97.8, 89.5, 64.2, 41.8, 57.2, 85, 91.4, 81.8, 74.6, 90]
print('No of observations : ',len(weight))
print('Sample Mean : ',np.mean(weight))
print('Standard Mean : ',np.std(weight,ddof=1))
ULT,LLT=stats.norm.interval(0.95,np.mean(weight),np.std(weight,ddof=1)/np.sqrt(1000))
print('Upper Limit : ',ULT,' Lower Limit:',LLT)

No of observations :  35
Sample Mean :  76.68571428571428
Standard Mean :  15.00909808116677
Upper Limit :  75.75545784270638  Lower Limit: 77.61597072872219


In [61]:
SD=190
ME=54
zscore=np.abs(round(stats.norm.isf(q=0.1/2),4))
print('Z-score:',zscore)
print('Sample size',round(zscore**2*SD**2/ME**2))

Z-score: 1.6449
Sample size 33


----------------------

**4.Problem:** . Hours spent on studies by 20 students on a course is given in the following table:**


|     |     |     |     |     |
| --- | --- | --- | --- | --- |   
| 4.7 | 9.2 | 9.3 | 11.2| 8   |
| 7.6 | 7.4 | 4.9 | 9.2 | 5.3 | 
| 1.7 | 2.8 | 7.2 | 12.3| 8.6 |
| 10.6| 9   | 5.7 | 6.9 | 3.8 |



**Assume that the population of hours spent follows a normal distribution and the standard deviation is 3.1 hours, calculate the 90% confidence interval for the mean hours spent by the students.**

In [62]:
import numpy as np
H           = [ 4.7, 9.2, 9.3, 11.2, 8, 7.6, 7.4, 4.9, 9.2, 5.3, 1.7, 2.8, 7.2, 12.3, 8.6, 10.6, 9, 5.7, 6.9, 3.8]
HoursSpent  = np.array(H)

In [76]:
mean=np.mean(HoursSpent)
print('Mean:',mean)
sd=3.1
stats.norm.interval(0.95,mean,sd/np.sqrt(len(HoursSpent)))

Mean: 7.2700000000000005


(5.9113900621063, 8.628609937893701)

**solution :**

In [66]:
sd=np.std(HoursSpent,ddof=1)
mean=np.mean(HoursSpent)
n=len(HoursSpent)
print('Sample size:',n)
print('SD',sd)
print('Mean',mean)
ULT,LMT=stats.norm.interval(0.9,mean,sd/np.sqrt(n))
print('Upper Limit: ',ULT,'Lower Limit',LMT)

Sample size: 20
SD 2.815203086999698
Mean 7.2700000000000005
Upper Limit:  6.234567031313146 Lower Limit 8.305432968686855


In [72]:
n=100
mean=35
sd=15
z_score=np.abs(round(stats.norm.isf(q=0.1/2),4))
print('Z_score',z_score)
ME=z_score*sd/np.sqrt(n)
print('Margin of error',ME)

Z_score 1.6449
Margin of error 2.46735


In [73]:
n=250
mean=45000
sd=4000
stats.norm.interval(0.95,mean,sd/np.sqrt(n))

(44504.16397415635, 45495.83602584365)

----------------------

**5.Problem**). The fill amount in 2-liter soft drink bottles is normally distributed, with a mean of 2.0 liters and a standard deviation of 0.05 liter. If the bottles contain less than 95% of the listed net content (1.90 liters, in our case), the manufacturer may be subject to penalty by the state office of consumer affairs. Bottles that have a net content above 2.1 liters may cause excess spillage upon opening. What is the proportion of bottles that will contain.**

* a) below 1.96 liters or above 2.2 liters
* b) At least how much soft drink is contained in 90% of the bottles?

**Solution:**

#### a. below 1.96 liters or above 2.2 liters is?
-----------

#### b) At least how much soft drink is contained in 90% of the bottles?
------------

---------------------

**10.Problem**). A sample of 100 diabetic patients was chosen to estimate the length of stay at a local hospital. 
The sample was 4.5 days and the population standard deviation was known to be 1.2 days.**

* a) Calculate the 99% confidence interval for the population mean.

**Solution**

a) Calculate the 99% confidence interval for the population mean.

It is known that 
* $\overline{X}$ = 4.5
* $\sigma$       = 1.2
* n              = 100
We need to compute $\overline{X} \quad \pm 1.64 \frac {\sigma} {\sqrt{n}}$ 

In [77]:
n=100
sd=1.2
mean=4.5
stats.norm.interval(0.99,mean,sd/np.sqrt(n))

(4.190900483574132, 4.809099516425868)

---------------------

**11.Problem**). Time taken to resolve a customer complaints in days of 100 customers in a Service Organization is given below:**

 |      |      |      |      |      |      |      |      |      |      |
 | ---  | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ---- |
 | 2.50 | 3.26 | 2.79 | 3.74 | 5.60 | 3.24 | 3.65 | 3.91 | 4.35 | 3.35 |
 | 5.67 | 5.38 | 3.54 | 5.10 | 3.66 | 3.01 | 3.96 | 4.98 | 4.56 | 5.00 |
 | 5.03 | 5.29 | 4.91 | 4.63 | 2.94 | 3.82 | 4.76 | 2.24 | 4.25 | 3.45 |
 | 3.14 | 4.64 | 4.56 | 4.61 | 2.68 | 3.61 | 5.46 | 2.83 | 4.84 | 4.31 |
 | 2.98 | 3.90 | 4.45 | 3.62 | 6.15 | 4.04 | 5.19 | 4.63 | 2.78 | 2.95 |
 | 3.65 | 4.49 | 3.52 | 4.07 | 4.16 | 5.56 | 2.69 | 6.69 | 1.26 | 3.14 |
 | 4.71 | 4.80 | 3.41 | 3.18 | 4.64 | 4.23 | 4.36 | 3.94 | 3.81 | 4.26 |
 | 2.92 | 2.87 | 2.08 | 3.09 | 3.60 | 2.93 | 3.85 | 4.66 | 4.70 | 3.61 |
 | 5.59 | 3.39 | 3.13 | 4.14 | 4.23 | 4.25 | 4.12 | 5.95 | 4.76 | 4.96 |
 | 2.27 | 3.77 | 5.25 | 3.05 | 3.20 | 5.22 | 3.84 | 2.24 | 4.75 | 3.07 |


* a. Construct 95% confidence interval estimate for the population mean days to resolve customer complaints,assuming a normal distribution.
* b. Interpret the interval constructed at a.

In [78]:
resolved_in_days = [2.50, 3.26, 2.79, 3.74, 5.60, 3.24, 3.65, 3.91, 4.35, 3.35,\
5.67, 5.38, 3.54, 5.10, 3.66, 3.01, 3.96, 4.98, 4.56, 5.00,\
5.03, 5.29, 4.91, 4.63, 2.94, 3.82, 4.76, 2.24, 4.25, 3.45,\
3.14, 4.64, 4.56, 4.61, 2.68, 3.61, 5.46, 2.83, 4.84, 4.31,\
2.98, 3.90, 4.45, 3.62, 6.15, 4.04, 5.19, 4.63, 2.78, 2.95,\
3.65, 4.49, 3.52, 4.07, 4.16, 5.56, 2.69, 6.69, 1.26, 3.14,\
4.71, 4.80, 3.41, 3.18, 4.64, 4.23, 4.36, 3.94, 3.81, 4.26,\
2.92, 2.87, 2.08, 3.09, 3.60, 2.93, 3.85, 4.66, 4.70, 3.61,\
5.59, 3.39, 3.13, 4.14, 4.23, 4.25, 4.12, 5.95, 4.76, 4.96,\
2.27, 3.77, 5.25, 3.05, 3.20, 5.22, 3.84, 2.24, 4.75, 3.07]


resolvedDays = np.array(resolved_in_days)

lengthStay = np.array(resolved_in_days)

avg        = lengthStay.mean(axis = 0)

sd         = np.std(lengthStay,ddof = 1)

print('Mean of length of stay is %1.4f' % avg)

print('SD of length of stay is %1.6f'   % sd)
# Here ddof modifies the divisor of the sum of the squares of the samples-minus-mean

Mean of length of stay is 4.0000
SD of length of stay is 1.000067


**Solution**

In [80]:
mean=np.mean(resolvedDays)
sd=np.std(resolvedDays)
print('Mean ',mean,' S.D',sd)
stats.norm.interval(0.95,mean,sd/np.sqrt(len(resolved_in_days)))

Mean  4.0  S.D 0.9950537673914913


(3.804973045323178, 4.1950269546768215)